### ISSUE TITLE
Glob-relevant characters in file paths are not escaped

### ISSUE THREAD

**Issue by chrisvfritz** [vuejs/eslint-config-typescript#102]
The generated paths for non-TS Vue files do not escape glob-relevant characters. For example:

```js
{
  name: 'typescript-eslint/disable-type-checked',
  files: [
    'src/pages/[...404].vue',
  ],
  // ...
}
```

In this case, `'src/pages/[...404].vue'` should be escaped as `'src/pages/\\[...404\\].vue'`.

**Comment by haoqunjiang** [comment]
Thanks!!! What an oversight!
I did know that the paths are actually globs, but forgot to test file-based routing.
